specification/synapse/data-plane/Microsoft.Synapse/stable/2020-12-01/entityTypes/Pipeline.json

```diff
@@ -7557,6 +7567,20 @@
         "modelAsString": true
       }
     },
+    "MetadataItem": {
+      "description": "Specify the name and value of custom metadata item.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "object",
```

Comment:
why `type` is `object`, not `string` ?

Reply:
ADF and Synapse support expressions, so type is object. It is our by design.
https://learn.microsoft.com/en-us/azure/data-factory/control-flow-expression-language-functions